### ISSUE TITLE
No such option: <some valid option>

### ISSUE THREAD

**Issue by stiege** [robotframework/robotframework#2173]
I have an untagged test inside a file which I want to run (but exclude some other tests)
```bash
pybot --exclude bootloaderORapplication -n noncritical .
```

However I get the output:
```robot
==============================================================================
Pu4-Test                                                                      
==============================================================================
Pu4-Test.Bootloader 2G4 Interface                                             
==============================================================================
The bootloader should have a radio interface 0x200000
Usage: pybot [options] [filename]

pybot: error: no such option: --exclude
[ ERROR ] Execution stopped by user.

```

As you can see, the tests begin, and the ` 0x200000` is a keyword executing which makes a command line call `st-info --flash` which writes that hex value to stdout.

Another person seemed to have a similar problem here:
http://stackoverflow.com/questions/31593743/no-such-option-include-for-pybot

```bash
pybot --version
Robot Framework 2.9.1 (Python 2.7.3 on linux2)

#Also reproduced after update:
pybot --version
Robot Framework 2.9.2 (Python 2.7.3 on linux2)
```

**Comment by stiege** [comment]
My special case seems to be that I was swapping an application over to being used as a library and python optparse was being used and raising the error, which is silly, but Robot is interpreting that error as its own I think, and so not generating any reports etc for any of the test cases.

**Comment by jussimalinen** [comment]
Sorry, I dont really understand what is going on here. What do you mean when you say that "you were swapping an application over to being used as a library"?

If you could provide an executable example that would replicate this issue, that would be great.

**Comment by stiege** [comment]
Hi, I mean the code I was working with was supposed to be called directly from the command line, i.e. it had a main() implementation and therefore used the optparse library to parse the command line arguments that were coming in.

I then modified it for the function to be called via another module, however it was still creating an OptionParser() and calling the parse_args() method.

I just looked into the source located at http://svn.python.org/projects/python/trunk/Lib/optparse.py . It looks like optparse prints the error and calls sysexit itself, so I think if that's the case there's nothing the Robot Framework could do to indicate that the developer is being a numpty.
